pynamodb/attributes.py

```diff
@@ -376,65 +348,42 @@ class BinarySetAttribute(SetMixin, Attribute[Set[bytes]]):
 
     def serialize(self, value):
         """
-        Returns a base64 encoded binary string
+        Returns a list of base64 encoded binary strings. Encodes empty sets as "None".
         """
-        if value and len(value):
-            return [b64encode(val).decode(DEFAULT_ENCODING) for val in sorted(value)]
-        else:
-            return None
+        return [b64encode(v).decode(DEFAULT_ENCODING) for v in value] if value else None
 
     def deserialize(self, value):
         """
-        Returns a decoded string from base64
+        Returns a set of decoded byte strings from base64 encoded values.
         """
-        try:
-            if value and len(value):
-                return {b64decode(val.decode(DEFAULT_ENCODING)) for val in value}
-        except AttributeError:
-            return {b64decode(val) for val in value}
+        return {b64decode(v) for v in value}
+
+
+class UnicodeAttribute(Attribute[str]):
+    """
+    A unicode attribute
+    """
+    attr_type = STRING
 
 
-class UnicodeSetAttribute(SetMixin, Attribute[Set[Text]]):
+class UnicodeSetAttribute(Attribute[Set[str]]):
     """
     A unicode set
     """
     attr_type = STRING_SET
     null = True
 
-    def element_serialize(self, value):
+    def serialize(self, value):
         """
-        This serializes unicode / strings out as unicode strings.
-        It does not touch the value if it is already a unicode str
-        :param value:
-        :return:
+        Returns a list of strings. Encodes empty sets as "None".
         """
-        if isinstance(value, str):
-            return value
-        return str(value)
-
-    def element_deserialize(self, value):
-        return value
-
-    def serialize(self, value):
-        if value is not None:
-            try:
-                iter(value)
-            except TypeError:
-                value = [value]
-            if len(value):
-                return [self.element_serialize(val) for val in sorted(value)]
-        return None
+        return list(value) if value else None
```

Comment:
Removes the sorting and the check for a non-iterable value. (The iteration check was already a bit broken since singleton-strings are utterable and would return a set of their characters.

Reply:
```suggestion
        return list(value) or None
```